src/NHibernate/Multi/IMultiAnyQuery.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Collections.Generic;
+using System.Data.Common;
+using NHibernate.Engine;
+using NHibernate.SqlCommand;
+
+namespace NHibernate
+{
+	/// <summary>
+	/// Interface for wrapping query to be batchable by <see cref="IMultiAnyQueryBatch"/>
+	/// </summary>
+	public interface IMultiAnyQuery<TResult> : IMultiAnyQuery
```

Comment:
Maybe rename it `IAnyQuery`. It is not itself multiple if I got it right. It is just there to supports query batch.